src/hooks/useCheckAccess.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useRef } from 'react';
+import { useLocation, useNavigate } from 'react-router';
+
+import useClientIntegration from './useClientIntegration';
+
+import { addQueryParam, removeQueryParam } from '#src/utils/location';
+import { checkEntitlements, reloadActiveSubscription } from '#src/stores/AccountController';
+
+type intervalCheckAccessPayload = {
+  interval?: number;
+  iterations?: number;
+  offerId?: string;
+};
+const useCheckAccess = () => {
+  const intervalRef = useRef<number>();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const intervalCheckAccess = ({ interval = 3000, iterations = 5, offerId }: intervalCheckAccessPayload) => {
+    const { clientOffers } = useClientIntegration();
+    if (!offerId && clientOffers?.[0]) {
```

Comment:
At the moment where the hook is used the offerId is not sent (even though there is a param for that), but checked from the config.. If we want it to be mandatory, we would still need to somehow get it from the config and pass it as an argument.. not sure if it will change anything.